File: HoYa/Entities/Quotation.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace HoYa.Entities
{

    public class Quotation : Detail<QuotationGeneral>//Process.Type.Value=="O"
    {
        public float? Amount { get; set; }
        public string Bargain { get; set; }
        public float? BargainPrice { get; set; }
        public virtual Guid? InquiryId { get; set; }
        [ForeignKey("InquiryId")]
        public virtual Inquiry Inquiry { get; set; }
    }


    public class QuotationGeneral : Base//Process.Type.Value=="O"
    {
        public virtual Guid? InquiryGeneralId { get; set; }
        [ForeignKey("InquiryGeneralId")]
        public virtual InquiryGeneral InquiryGeneral { get; set; }
    }
}

File: HoYa.Web/src/entities/quotation.ts
import { Detail, Base } from "./entity";
import { InquiryGeneral, Inquiry } from "./inquiry";

export class Quotation extends Detail<QuotationGeneral>
{
    amount: number;
    bargain: string;
    bargainPrice: number;
    inquiryId: string;
    inquiry: Inquiry;
    constructor() {
        super();
    }
}


export class QuotationGeneral extends Base {
    inquiryGeneralId: string;
    inquiryGeneral: InquiryGeneral;
    constructor() {
        super();
    }
}